### PR TITLE
feat: added a `description` field to instructions

### PR DESCRIPTION
**Pull request by h4ck3rk3y**
## Description

The description field can be used instead of the full instruction while outputting to the end user

In the CLI this can be accessed using `--description`;

### CODE REVIEW

File: core/server/api_container/server/startosis_engine/kurtosis_instruction/upload_files/upload_files.go [hunk 1]
```diff
@@ -213,3 +213,7 @@ func (builtin *UploadFilesCapabilities) FillPersistableAttributes(builder *encla
 		builtin.artifactName, builtin.filesArtifactMd5,
 	)
 }
+
+func (builtin *UploadFilesCapabilities) Description() string {
+	return fmt.Sprintf("Uploading file '%v' to files articact '%v'", builtin.src, builtin.artifactName)
```
Comment (h4ck3rk3y) [inline core/server/api_container/server/startosis_engine/kurtosis_instruction/upload_files/upload_files.go:218]: @mieubrisse if you can go through just the instructions and see if you are happy with the description per instruction

**Review by leoporoli — APPROVED** [review]

File: core/server/api_container/server/startosis_engine/kurtosis_instruction/upload_files/upload_files.go (hunk 1 above)
Comment (h4ck3rk3y) [reply core/server/api_container/server/startosis_engine/kurtosis_instruction/upload_files/upload_files.go:218]: will merge current iteration tonight if no comments we can correct later

Comment (mieubrisse) [inline core/server/api_container/server/startosis_engine/kurtosis_instruction/upload_files/upload_files.go:218]: s/articact/artifact/

File: core/server/api_container/server/startosis_engine/kurtosis_instruction/verify/verify.go [hunk 2]
```diff
@@ -155,6 +155,10 @@ func (builtin *VerifyCapabilities) FillPersistableAttributes(builder *enclave_pl
 	builder.SetType(VerifyBuiltinName)
 }
 
+func (builtin *VerifyCapabilities) Description() string {
+	return fmt.Sprintf("Verifying whether two values meet a certain condition '%v'", builtin.assertion)
```
Comment (mieubrisse) [inline core/server/api_container/server/startosis_engine/kurtosis_instruction/verify/verify.go:159]: Is it possible to add a bit more info on what those values are?

File: core/server/api_container/server/startosis_engine/kurtosis_instruction/wait/wait.go [hunk 3]
```diff
@@ -288,3 +288,7 @@ func (builtin *WaitCapabilities) TryResolveWith(instructionsAreEqual bool, _ *en
 func (builtin *WaitCapabilities) FillPersistableAttributes(builder *enclave_plan_persistence.EnclavePlanInstructionBuilder) {
 	builder.SetType(WaitBuiltinName)
 }
+
+func (builtin *WaitCapabilities) Description() string {
+	return fmt.Sprintf("Waiting for at most '%v' for service '%v' to reach a certain state", builtin.timeout, builtin.serviceName)
```
Comment (mieubrisse) [inline core/server/api_container/server/startosis_engine/kurtosis_instruction/wait/wait.go:293]: Same here; can we give a bit more info on what the certain state is?

File: core/server/api_container/server/startosis_engine/kurtosis_instruction/tasks/run_sh.go [hunk 4]
```diff
@@ -266,6 +266,10 @@ func (builtin *RunShCapabilities) FillPersistableAttributes(builder *enclave_pla
 	builder.SetType(RunShBuiltinName)
 }
 
+func (builtin *RunShCapabilities) Description() string {
+	return "Running a one time bash script"
```
Comment (mieubrisse) [inline core/server/api_container/server/startosis_engine/kurtosis_instruction/tasks/run_sh.go:270]: Could we make this `Running sh script` in the case where there's a lot of command to run, and `Running sh script: .....` if not?

File: core/server/api_container/server/startosis_engine/kurtosis_instruction/tasks/run_python.go [hunk 5]
```diff
@@ -321,6 +321,10 @@ func (builtin *RunPythonCapabilities) FillPersistableAttributes(builder *enclave
 	builder.SetType(RunPythonBuiltinName)
 }
 
+func (builtin *RunPythonCapabilities) Description() string {
+	return "Running a one time python script"
```
Comment (mieubrisse) [inline core/server/api_container/server/startosis_engine/kurtosis_instruction/tasks/run_python.go:325]: Can we make this `Running Python script`? (also just a reminder that for consistency we use proper-casing for proper nouns like Python, Docker, Kubernetes, etc.)

File: core/server/api_container/server/startosis_engine/kurtosis_instruction/store_service_files/store_service_files.go [hunk 6]
```diff
@@ -171,3 +171,7 @@ func (builtin *StoreServiceFilesCapabilities) FillPersistableAttributes(builder
 		builtin.artifactName, nil,
 	)
 }
+
+func (builtin *StoreServiceFilesCapabilities) Description() string {
+	return fmt.Sprintf("Storing files from service '%v' at path '%v' to files artifact with name '%v'", builtin.serviceName, builtin.src, builtin.artifactName)
```
Comment (mieubrisse) [inline core/server/api_container/server/startosis_engine/kurtosis_instruction/store_service_files/store_service_files.go:176]: Nice; like this one

**Review by mieubrisse — COMMENTED** [review]
Added some suggestions!

**Pull request merged**